### PR TITLE
Fix appBoot and firstMessage whisper bug

### PR DESCRIPTION
**Pull request by lemieux**
There was a bug in the way the response from the server was handling cause a `track` to never be successful (the server got the info and all but the client always had an error).

### CODE REVIEW

File: src/js/utils/api.js [hunk 1]
```diff
@@ -26,6 +26,10 @@ function status(response) {
 }
 
 function json(response) {
+    if (response.status === 202 || response.status === 204) {
```
Comment (mspensieri) [inline src/js/utils/api.js]: what does `response.json()` return if not `undefined` or `null`? Or does it throw?

Comment (lemieux) [inline src/js/utils/api.js]: Pretty sure it was throwing, but now I'm just trying to test it again and I can't figure a call we do an end up with a 204

**Comment by lemieux** [comment]
After testing with @Mario54 , this PR doesn't fix the first user message bug at all. I'll have to work on that.

File: src/js/utils/api.js (hunk 1 above)
Comment (lemieux) [inline src/js/utils/api.js]: Yeah it was throwing. I tested it with tracking an event for a whisper.

File: src/js/controllers/chatController.js [hunk 2]
```diff
@@ -197,17 +203,14 @@ module.exports = ViewController.extend({
     _initConversation: function() {
         var promise;
 
-        if (this.conversationInitiated) {
+        if (this.collection.length > 0 && !this.collection.at(0).isNew()) {
```
Comment (lemieux) [inline src/js/controllers/chatController.js]: This is the key change that fixed the first message disappearance. It relied on `this.conversationInitiated` that was set to true further down the promise chain. I'm pretty there was a race condition when we tracked an event and/or update the user and the conversation is started there too.

**Comment by lemieux** [comment]
@Mario54 tested it and the bug seems to be gone. I was able to reproduce it very easily and it seems to be gone too.

**Comment by spasiu** [comment]
:+1:

**Comment by mspensieri** [comment]
:+1:

**Pull request merged**